src/classes/RD2_DataMigration_TEST.cls

```diff
@@ -1855,6 +1855,60 @@ private class RD2_DataMigration_TEST {
             'Number of Planned Installments should be set to null');
     }
 
+    /**
+     * @description Verifies if the Accounts are populated after data migration is run
+     * when household contact is only populated and account is null.
+     */
+    @IsTest
+    private static void shouldPopulateAccountWhenContactIsOnlyEntered() {
+        npe03__Recurring_Donation__c rd = getLegacyRecurringDonationBuilder(getContact().Id)
+            .withId(null)
+            .withAccount(null)
+            .build();
+        insert rd;
+
+        Id jobId = runDataMigration();
+
+        npe03__Recurring_Donation__c convertedRD = getRecords()[0];
```

Comment:
For this test and the one below it:
- No need to create a new var, can reuse the `rd`.
- Assert message can be simplified.
```
        rd = getRecords()[0];
        System.assertNotEquals(null, rd.npe03__Organization__c,
             'Account on RD should be populated: ' + rd);
        System.assertEquals(getContact().AccountId, rd.npe03__Organization__c,
             'Account on RD should be the Account associated with the Contact: ' + rd);
```